### PR TITLE
Change step definition structure

### PR DESCRIPTION
**Pull request by robertocgongora**
[Trello Card](https://trello.com/c/rRbvaA7G/1-change-step-definition-structure)

# Overview

* moved steps definitions to: login.steps.js, taskCreation.steps.js and hooks.js
* updated feature files(removed ":")
* added some RegEx in steps 

## Dependencies
- None

### CODE REVIEW

File: test/e2e/steps/login.steps.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const LoginAssert = require('../asserts/Login.assert');
+const LoginPage = require('../pages/LoginPage');
+const { Given, When, Then } = require('cucumber');
+
+Given(/^user navigates to "([^"]*)"$/, (endpoint) => {
```
Comment (miv1) [inline test/e2e/steps/login.steps.js:5]: According the article https://automationpanda.com/2017/01/30/bdd-101-writing-good-gherkin/ Given step should be a status, not an action

Comment (robertocgongora) [reply test/e2e/steps/login.steps.js:5]: yep, totally agree, please report it as a bug, this code was defined before the movements in steps definitions. In this case, this step was defined by @R0m3rCh

File: test/e2e/steps/taskCreation.steps.js [hunk 2]
```diff
@@ -74,11 +67,17 @@ When('user adds description to a given card:', function (dataTable) {
...
     TaskPage.taskSettingsButton.click();
     TaskPage.deleteButton.click();
   });
 });
+
+Then(/^the following tasks should be visible$/, function (dataTable) {
+  let tasks = dataTable.raw();
+  tasks.forEach(elm => TaskAsserts.assertTaskExist(elm));
+});
+
```
Comment (miv1) [inline test/e2e/steps/taskCreation.steps.js]: No new line at EOF

File: test/e2e/steps/taskCreation.steps.js [hunk 3]
```diff
@@ -1,56 +1,49 @@
...
-
-When('user closes the first task', function () {
+  
+When(/^user closes the first task$/, function () {
   TaskPage.taskTitle.waitForExist(3000);
   TaskPage.taskTitle.moveTo();
   TaskPage.closeTaskIcon.click();
 });
-
-When('user creates {string} board', function (boardName) {
```
Comment (R0m3rCh) [inline test/e2e/steps/taskCreation.steps.js:43]: This step hasn't made it to the final structure

Comment (robertocgongora) [reply test/e2e/steps/taskCreation.steps.js:43]: yep, you're right, let me add this step definition

Comment (robertocgongora) [reply test/e2e/steps/taskCreation.steps.js:43]: DONE

File: test/e2e/steps/taskCreation.steps.js (hunk 2 above)
Comment (robertocgongora) [reply test/e2e/steps/taskCreation.steps.js]: DONE

File: test/e2e/steps/taskCreation.steps.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+});
+    
+When(/^user creates tasks with the following information$/, function (dataTable) {
+  let task = dataTable.hashes();
+  for(let i = 0; i < task.length; i++) {
+    DashboardPage.setTaskName(task[i].taskName);
+    DashboardPage.saveTaskButton.click();
+  }
+});
+When(/^user clicks on BOARD tab$/, function () {
```
Comment (miv1) [inline test/e2e/steps/taskCreation.steps.js]: Please, leave a blank space at the end of the step

Comment (robertocgongora) [reply test/e2e/steps/taskCreation.steps.js]: DONE

File: test/e2e/steps/taskCreation.steps.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+});
+
+When(/^user creates "([^"]*)" board$/, function (boardName) {
+    DashboardPage.setTaskName('      ');
+    DashboardPage.boardTab.click();
+    DashboardPage.addStatusButton.click();
+    BoardPage.setStatusNameInput(boardName);
+    browser.keys(SystemInteractions.ENTER_KEY_PRESS);
+    DashboardPage.modalText.click();
+    DashboardPage.confirmNewStatus.click();
```
Comment (R0m3rCh) [inline test/e2e/steps/taskCreation.steps.js]: There are extra tabulations here

**Review by miv1 — APPROVED** [review]

File: test/e2e/steps/taskCreation.steps.js (hunk 5 above)
Comment (robertocgongora) [reply test/e2e/steps/taskCreation.steps.js]: Damn! you're right, I completely forgot that we are working with 2 spaces, great catch!!

Comment (robertocgongora) [reply test/e2e/steps/taskCreation.steps.js]: DONE

**Review by R0m3rCh — APPROVED** [review]

**Pull request merged**